frontend/src/utils/repo-sorter.ts

```diff
@@ -1,12 +1,27 @@
-function getTotalCommits(total, group) {
-  return total + group.checkedFileTypeContribution;
+import { User } from '../types/types';
+import { FilterGroupSelection } from '../types/summary';
+
+function getTotalCommits(total: number, group: User) {
+  // If group.checkedFileTypeContribution === undefined, then we treat it as 0 contribution
+  return total + (group.checkedFileTypeContribution ?? 0);
 }
 
-function getGroupCommitsVariance(total, group) {
+function getGroupCommitsVariance(total: number, group: User) {
   return total + group.variance;
 }
 
-function sortingHelper(element, sortingOption) {
+function checkKeyAndGetValue(element: object, key: string | undefined) {
+  if (key === undefined || !(key in element)) {
+    return undefined;
+  }
+  /** Casting here is safe as guard clause above ensures that element has an attribute with the same name as the string
+   *  represented by key. */
+  return (element as any)[key];
```

Comment:
See if it is possible to use generics here. 
Else disable eslint for this line using `// eslint-disable-next-line  @typescript-eslint/no-explicit-any`.

Reply:
Thanks for the review @ckcherry23, I have experimented with making use of generics over here and it seems like there is no real need for generics (although I have left a function that uses generics for general use). All that is needed would be casting the `key` string into a `keyof User`. However, this causes the problem of handling invalid keys and values that are not `ComparablePrimitive`, i.e. `string | number` to resurface (it was being suppressed due to the `any` typecast previously used). I have tried to handle these cases, but am unsure whether such behavior is appropriate (details below). Do let me know if the current behavior is not as intended.

### Details of behavior
When a `key` that is not valid or simply if a key is not provided to the function, `checkKeyAndGetValue` will return undefined, which will be caught in the `getComparablePrimitive` function's guard clause, resulting in an empty string being returned.
When a `key` that is associated with a value that is not a `ComparablePrimitive`, i.e. not a `string | number`, is provided to the function, it is caught in the `getComparablePrimitive` function's guard clause, also resulting in an empty string being returned.
When an empty string is returned, this results in the associated `User` being sorted to the front. If all `User`s in the `User[]`  result in an empty string, no sorting occurs.